aten/src/ATen/native/PackedSequence.cpp

```diff
@@ -67,8 +72,6 @@ std::tuple<Tensor, Tensor> _pack_padded_sequence(const Tensor& _input, const Ten
         (*batch_sizes++) = current_batch_size;
       }
       prev_l = l;
-    } else if (prev_l > l) {
```

Comment:
You don't need to check that l > prev_l anymore. Could you add an assert (AT_ASSERT(l > prev_l)) ?